src/controllers/update/overseas.entity.query.controller.ts

```diff
@@ -32,29 +34,17 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
 
     const entityNumber = req.body[EntityNumberKey];
     const companyProfile = await getCompanyProfile(req, entityNumber);
+
     if (!companyProfile) {
-      const errors = createEntityNumberError(entityNumber);
-      return res.render(config.OVERSEAS_ENTITY_QUERY_PAGE, {
-        backLinkUrl: config.UPDATE_LANDING_PAGE_URL,
-        templateName: config.OVERSEAS_ENTITY_QUERY_PAGE,
-        [EntityNumberKey]: entityNumber,
-        errors
-      });
-    } else {
-      const appData: ApplicationData = getApplicationData(req.session);
-      if (appData.entity_number !== entityNumber) {
-        resetEntityUpdate(appData);
```

Comment:
We need to change resetEntityUpdate to also do the rest on review lists:

appData.update.bo_mo_data_fetched = false;

  appData.update.review_beneficial_owners_individual = [];
  appData.update.review_beneficial_owners_corporate = [];
  appData.update.review_beneficial_owners_government_or_public_authority = [];
  appData.update.review_managing_officers_individual = [];
  appData.update.review_managing_officers_corporate = [];

Reply:
That's not a new method that's been added as part of this ticket, also looks like it already resets appData.update.

`export const resetEntityUpdate = (appData: ApplicationData): Update => {
  // Entity data will be filled in by caller so not reset.
  appData.update = {};`

Reply:
You're right I was concerned bo_mo_data_fetched was not explicitly being reset.